Xamarin.Forms.Core/GestureRecognizer.cs

```diff
@@ -1,9 +1,222 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
 namespace Xamarin.Forms
 {
 	public class GestureRecognizer : Element, IGestureRecognizer
 	{
+		public static readonly BindableProperty TouchStateProperty =
+			BindableProperty.Create(nameof(TouchState), typeof(TouchState), typeof(GestureRecognizer), TouchState.Default, BindingMode.OneWayToSource);
+
+		public static readonly BindableProperty TouchCountProperty =
+			BindableProperty.Create(nameof(TouchCount), typeof(int), typeof(GestureRecognizer), 0, BindingMode.OneWayToSource);
+
+		public static readonly BindableProperty TouchesProperty =
+			BindableProperty.Create(nameof(Touches), typeof(List<Touch>), typeof(GestureRecognizer), null, BindingMode.OneWayToSource);
+
+		readonly Dictionary<int, Touch> _touches = new Dictionary<int, Touch>();
+
 		internal GestureRecognizer()
 		{
 		}
+
+		public TouchState TouchState
+		{
+			get => (TouchState)GetValue(TouchStateProperty);
+			private set => SetValue(TouchStateProperty, value);
+		}
+
+		public int TouchCount
+		{
+			get => (int)GetValue(TouchCountProperty);
+			private set => SetValue(TouchCountProperty, value);
+		}
+
+		public List<Touch> Touches
+		{
+			get => (List<Touch>)GetValue(TouchesProperty);
+			private set => SetValue(TouchesProperty, value);
+		}
```

Comment:
if the setters are private, you should use BindablePropertyKeys, to make ro-BP

Reply:
done